component/file_cache/file_cache_test.go

```diff
@@ -1634,8 +1645,35 @@ func (suite *fileCacheTestSuite) TestHardLimitOnSize() {
 	suite.assert.Error(err)
 }
 
+func (suite *fileCacheTestSuite) TestHandleDataChange() {
+	defer suite.cleanupTest()
+	// Configure to create empty files so we create the file in cloud storage
+	createEmptyFile := true
+	config := fmt.Sprintf("file_cache:\n  path: %s\n  offload-io: true\n  create-empty-file: %t\n  timeout-sec: 1000\n  refresh-sec: 10\n\nloopbackfs:\n  path: %s",
+		suite.cache_path, createEmptyFile, suite.fake_storage_path)
+	suite.setupTestHelper(config) // setup a new file cache with a custom config (teardown will occur after the test as usual)
+
+	path := "file43"
+	err := os.WriteFile(suite.fake_storage_path+"/"+path, []byte("test data"), 0777)
+	suite.assert.NoError(err)
+
+	data := make([]byte, 20)
+	options := internal.OpenFileOptions{Name: path, Flags: os.O_RDONLY, Mode: 0777}
+
+	// Read file once and we shall get the same data
+	f, err := suite.fileCache.OpenFile(options)
+	handlemap.Add(f)
+	suite.assert.NoError(err)
+	suite.assert.False(f.Dirty())
+	n, err := suite.fileCache.ReadInBuffer(internal.ReadInBufferOptions{Handle: f, Offset: 0, Data: data})
+	f, loaded := handlemap.Load(f.ID)
+	suite.assert.True(loaded)
+	suite.assert.NoError(err)
+	suite.assert.Equal(9, n)
+	err = suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: f})
+	suite.assert.NoError(err)
+
+}
+
 // In order for 'go test' to run this suite, we need to create
 // a normal test function and pass our suite to suite.Run
-func TestFileCacheTestSuite(t *testing.T) {
```

Comment:
Why did you remove this? This is needed for the test suite to run. Right now these tests are not running in the test suite.

Reply:
I don't know how that happened. I'll put it back